### ISSUE TITLE
Function hooking

### ISSUE THREAD

**Issue by vars1ty** [veeenu/hudhook#118]
Have you considered using https://crates.io/crates/retour instead of MinHook? It's essentially a maintained version of "detours".
Should also be easier to use than MH.

**Comment by veeenu** [comment]
Hi!

Yeah, I've seen it. I decided against it because we don't need anything more than what `minhook` already offers, plus `minhook` is very minimal and stable, and its API is clean and makes FFI not really a problem at all.

On the other hand, `hudhook`'s dependency tree has a history of small packages ending up abandoned in a broken state, someone forks them and the fork becomes abandoned as well shortly after. I actually empathize big time: maintaining stuff is hard, and this very repo and the paltry progress I've been able to do on it recently is a testament to that. 😅 

`detours` is one of the crates that ended up not being maintained, and, while `retour` looks promising, switching over entails a maintenance burden that doesn't justify the value added to our use case.
